### PR TITLE
Support Popup Translations

### PR DESCRIPTION
**Pull request by TurtIeSocks**
- All popups are now translatable
- Some additional menus as well
- Changed most popups to use `` instead of ''
- Translate Rarity Tiers
- Translate help menu
- Translate weather and boosted types

### CODE REVIEW

**Review by lenisko — CHANGES_REQUESTED** [review]
one wrongly picked key

File: static/js/index.js [hunk 1]
```diff
@@ -3616,34 +3616,34 @@ function getGymPopupContent (gym) {
...
             if (gym.raid_pokemon_move_2 !== null) {
                 let move2 = getMoveName(gym.raid_pokemon_move_2);
                 if (move2 !== null) {
-                    content += '<b>Charge:</b> ' + move2 + '<br>';
+                    content += `<b>${i18n('popup_charge')}:</b> ${move2}<br>`;
                 }
             }
             if (gym.in_battle) {
-                content += '<b>Gym last seen in battle!</b><br>';
+                content += `<b>${i18n('gym_last_seen_in_battle')}!</b><br>`;
```
[the remark this answers is not in the document]
Comment (TurtIeSocks) [reply static/js/index.js]: Resolved in following commit, thanks!

**Review by versx — APPROVED** [review]
Hell yeah, nice job!

File: src/views/index.mustache [hunk 2]
```diff
@@ -356,38 +356,38 @@
...
+                    <div class="help-title"><b>{{general_basics}}</b></div>
                     <div class="help-description">
                         <p>
-                            IV Range: <kbd>90-100</kbd> - 90-100% IV Pokemon<br>
-                            Level Range: <kbd>L30-35</kbd> - Level 30 to 35 Pokemon<br>
-                            Stats: <kbd>A0-1 & D14 & S15</kbd> - Pokemon with 0-1 Attack, 14 Defense & 15 Stamina<br>
-                            CP: <kbd>CP2000-3000</kbd> - All Pokemon with CP of 2000-3000<br>
-                            NOT: <kbd>!0-100</kbd> - All Pokemon that do not have their IVs scanned<br>
-                            PVP: <kbd>GL1-3 | UL1-3</kbd> - Pokemon that are rank 1-3 for either Great League or Ultra League<br><br>
+                            {{iv_range}}: <kbd>90-100</kbd> - {{iv_example}}<br>
```
Comment (lenisko) [inline src/views/index.mustache:364]: `iv_example` is a bit confusing as a name as it's not an example. What about `iv_example_help` or `iv_explanation` ? (same for others).

File: static/locales/_en.json [hunk 3]
```diff
@@ -280,15 +280,75 @@
...
+    "level_example": "Level 30 to 35 Pokemon",
+    "stats": "Stats",
+    "stats_example": "Pokemon with 0-1 Attack, 14 Defense, & 15 Stamina",
+    "cp_example": "All Pokemon with CP of 2000-3000",
+    "not": "NOT",
+    "not_example": "All Pokemon that do not have their IVs scanned",
+    "pvp": "PVP",
+    "pvp_example": "Pokemon that are rank 1-3 for either Great League or Ultra League",
+    "stringing_conditions_together": "Stringing Conditions Together",
+    "or": "or",
```
Comment (lenisko) [inline static/locales/_en.json:346]: `NOT` and help strings logic gates are already upper case. Lets make them all upper.

Small bonus with `On` and `Off`. 

It's much easier to read them like that I think.
```
    "filter_and": "And",
    "filter_or": "Or",
    "filter_on": "On",
    "filter_off": "Off",
    "or": "or",
```

**Review by lenisko — CHANGES_REQUESTED** [review]

**Review by lenisko — APPROVED** [review]

**Review by versx — APPROVED** [review]

**Review by versx — APPROVED** [review]

**Review by versx — APPROVED** [review]

**Comment by lenisko** [comment]
So important that @versx approved it 3 times in row! (and 4 in total 😂)

**Pull request merged**